projects/components/src/datagrid/details/detail-wrapper.component.ts

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { Component, Injector, Input } from '@angular/core';
+import { LoadingListener } from '@clr/angular';
+import { ComponentRendererConstructor, ComponentRendererSpec, DetailRowConfig } from '..';
+
+/**
+ * A component that wraps the detail row to provide LoadingListener functionality.
+ *
+ * This component only exists as an easy way to get access to the {@link LoadingListener}.
+ */
+@Component({
+    selector: 'vcd-datagrid-detail-wrapper',
+    templateUrl: './detail-wrapper.component.html',
+})
+export class DatagridDetailWrapperComponent<R> {
+    injector: Injector;
+    /**
+     * The record that this row detail will display.
+     */
+    @Input() record: R;
+    /**
+     * The index this row detail is located at.
+     */
+    @Input() index: number;
+    /**
+     * The total number of rows in the grid.
+     */
+    @Input() count: number;
+    /**
+     * The component that sound be rendered for this detail row.
+     *
+     * @param R The type of record that this detail component will display.
+     */
+    @Input() detailComponent: ComponentRendererConstructor<DetailRowConfig<R>>;
+
+    /**
+     * Constructs this detail wrapper with the injected loading listener.
+     */
+    constructor(public loadingListener: LoadingListener) {
```

Comment:
If we instead do

```ts
constructor(public injector:Injector) { }
```

Would that allow the custom detail component to access the rest manager in vcd_ui?

Right now it looks like the custom detail component wouldn't have access to anything it could need through DI